### PR TITLE
feat: restrict mutable globals to be export in gear

### PR DESCRIPTION
**Pull request by grishasobol**
restrict mutable globals to be export in gear on instrumentation. Exception is GEAR_STACK_END but only temporary.  In nearest feature GEAR_STACK_END will point to self const export.

### CODE REVIEW

File: pallets/gear-program/src/mock.rs [hunk 1]
```diff
@@ -40,8 +40,11 @@ type AccountId = u64;
 type BlockNumber = u64;
 type Balance = u128;
 
+#[allow(unused)]
```
Comment (breathx) [inline pallets/gear-program/src/mock.rs:43]: Annoys me too. Better to `#![allow(unused)]` whole mock mod, because it's only test tooling

Comment (grishasobol) [reply pallets/gear-program/src/mock.rs]: Done

**Review by NikVolf — APPROVED** [review]
As CI seems fine, Rust does not produce public mutable globals in the binaries?

File: core/src/code.rs [hunk 2]
```diff
@@ -202,6 +208,44 @@ pub struct Code {
     instruction_weights_version: u32,
 }
 
+fn check_for_mut_global_exports(module: &Module) -> Result<(), CodeError> {
+    let global_exports_indexes = module
```
Comment (NikVolf) [inline core/src/code.rs:212]: please no, don't put it in the protocol 

just make it const in instrumentantion or whatever

**Review by NikVolf — CHANGES_REQUESTED** [review]
shouldn't put magic global names in the protocol

**Comment by grishasobol** [comment]
> As CI seems fine, Rust does not produce public mutable globals in the binaries?
Yes

File: core/src/code.rs (hunk 2 above)
Comment (grishasobol) [reply core/src/code.rs:212]: Do you mean make __gear_stack_end export to const global?

**Comment by grishasobol** [comment]
> shouldn't put magic global names in the protocol

This const is already in protocol since 2022 February =)

**Review by NikVolf — APPROVED** [review]
other than that make this injected stack pointer non-mutable

File: core/src/code.rs (hunk 2 above)
Comment (grishasobol) [reply core/src/code.rs:258]: Done

**Review by ukint-vs — APPROVED** [review]

**Review by gshep — APPROVED** [review]
💯

**Pull request merged**